test_scripts/API/VehicleData/WindowStatus/023_OnVehicleData_notification_with_invalid_data.lua

```diff
@@ -0,0 +1,143 @@
+---------------------------------------------------------------------------------------------------
+-- Proposal:https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0261-New-vehicle-data-WindowStatus.md
+--
+-- Description: Check that SDL does not transfer OnVehicleData notification to subscribed app if HMI sends notification
+-- with invalid values of `windowStatus` structure params:
+--    location: { col, row, level, colspan, rowspan, levelspan }
+--    state: { approximatePosition, deviation }
+--
+-- In case:
+-- 1) App is subscribed to `windowStatus` data.
+-- 2) HMI sends the invalid `windowStatus` structure in OnVehicleData notification:
+--    - invalid parameter name
+--    - invalid parameter type
+--    - missing mandatory parameter
+--    - value out of bounds
+--    - empty structures
+-- SDL does:
+--  a) ignore this notification.
+--  b) not send OnVehicleData notification to mobile.
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require('test_scripts/API/VehicleData/WindowStatus/common')
+
+-- [[ Test Configuration ]]
+config.zeroOccurrenceTimeout = 500
+
+--[[ Local Variables ]]
+local invalidValue = {
+  invalidType = true,
+  greaterMax = 101,
+  lessThanMin = -2
+}
+
+local maxArraySize = {}
+for i = 1, invalidValue.greaterMax do
+  maxArraySize[i] = common.getWindowStatusParams()[1]
+end
+
+local invalidParam = {
+  ["empty_location"] = {
```

Comment:
@GetmanetsIrina , please add checks with missed approximatePosition or deviation.

Reply:
and also add checks for GetVD RPC

Reply:
@atiwari9, the update according to comments is performed in 3092bef5f0f3b3870272195f009bd15a5c0a8501